data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -170,8 +226,23 @@ public void closePartition(final String partitionKey, final Duration reopenAfter
             itemToUpdate.setReOpenAt(Instant.now().plus(reopenAfter));
         }
 
+        try {
+            sourceCoordinationStore.tryUpdateSourcePartitionItem(itemToUpdate);
+        } catch (final PartitionUpdateException e) {
+            if (SourcePartitionStatus.COMPLETED.equals(itemToUpdate.getSourcePartitionStatus())) {
+                completePartitionUpdateErrorCounter.increment();
+            } else {
+                closePartitionUpdateErrorCounter.increment();
```

Comment:
This logic won't extend well if we add a new status that is supported in this method. SaveProgressState does not have a metric for each partition status type. Do we need to differentiate between Closed and Completed? I would recommend creating a dimension that keyed off of status if we need to.

Reply:
The only updates are saveState, complete, and close. I chose to give granularity on which action failed to update by separating to three different metrics. I am fine with adding a single metric for `partitionUpdateErrors` with a dimension that shows the action, but how does this work with micrometer